install/agave-install-init.sh

```
@@ -50,6 +50,7 @@ main() {
     need_cmd rm
     need_cmd sed
     need_cmd grep
+    need_cmd rustup
```

Given that we're now telling devs that need to install `rustup`, should this still be here ?

---

evne thought the docs state to install it, if a dev does not have it installed the platform tools code will silently fail to create the `solana` toolchain. having the command check prevents it

---

Can the installer still be run successfully if `rustup` is not installed ?

---

It will error if rustup is not available. Do you want me to update this script to install rustup if it is not available (by running the rustup installer shell command listed on the rustup.rs website)?

---

> It will error if rustup is not available

Got it; I should have looked to see what `need_cmd` does myself 😅 

> Do you want me to update this script to install rustup if it is not available (by running the rustup installer shell command listed on the rustup.rs website)?

My initial thought is no, but I can ping someone else from Anza to weigh in on if I'm being too paranoid.

Back to the two groups thought:
1. Developers probably want the tools, so they probably want an error or warning that will tell them if the tools won't be properly installed
2. Regular users (ie person X who is native staking) don't want or need the tool, so they don't want or need the dependency (`rustup`) for the tools either

So, group 1 wants a blocking error seemingly whereas group 2 wants a warning at the very most. These thoughts are seemingly at odds with each

---

no. do not auto install anything ever. user needs to consent to changes on their system and there's no value to maintaining such logic